### PR TITLE
removed some async references

### PR DESCRIPTION
**Pull request by ajb413**

### CODE REVIEW

File: src/components/search.js [hunk 1]
```diff
@@ -188,7 +190,26 @@ class Search extends Emitter {
                     response.messages.reverse();
                 }
 
-                eachSeries(response.messages, this.triggerHistory, () => {
+                return new Promise((resolve, reject) => {
+
+                    let iteration = 0;
+                    let collection = response.messages;
+
+                    let forEachSeries = (iteratee, toRun) => {
```
Comment (ianjennings) [inline src/components/search.js]: Is this really better? This was my train of thought:

* lets get rid of aysnc/eachSeries
* ok how do you do this in ES6
* I go to stack overflow
* That's a handy function
* This seems like a utility function that should be defined somewhere else
* Hrm, ok I should define this in another file
* Wait, isn't defining the utility function in another file conceptually the same as ```require('async/eachSeries')```

File: src/components/chat.js [hunk 2]
```diff
@@ -601,52 +600,33 @@ class Chat extends Emitter {
...
-                    })
-                    .catch(next);
-
-            }
-        ], (error) => {
-            this.chatEngine.throwError(this, 'trigger', 'auth', new Error('Something went wrong while making a request to authentication server.'), { error });
-        });
+            })
+            .catch((error) => {
+                this.chatEngine.throwError(this, 'trigger', 'auth', new Error('Something went wrong while making a request to authentication server.'), { error });
```
Comment (ianjennings) [inline src/components/chat.js]: That's nice but we should test it as we chatted about.

File: src/components/search.js (hunk 1 above)
Comment (ianjennings) [reply src/components/search.js]: I'm not saying don't do this, I'm saying that using a library over a custom utility may be a better bet and may not necessarily be larger.

**Comment by ianjennings** [comment]
With my updates you're at 128kb minified. We started at 158kb and #246 is at 136kb. 


#247 - 19% Reduction
#246 - 14% Reduction

File: src/bootstrap.js [hunk 3]
```diff
@@ -394,30 +392,19 @@ module.exports = (ceConfig = {}, pnConfig = {}) => {
...
-            },
-            (next) => {
-                ChatEngine.request('post', 'group').then(complete).catch(next);
-            }
-        ], (error) => {
-            if (error) {
-                ChatEngine.throwError(ChatEngine, '_emit', 'auth', new Error('There was a problem logging into the auth server (' + ceConfig.endpoint + ').'), { error });
-            }
-        });
+        ChatEngine.request('post', 'bootstrap').then(() => {
```
[the remark this answers is not in the document]
Comment (ajb413) [reply src/bootstrap.js]: at first I tried parallel and the tests returned a lot of 404s. making it series fixed it.

[the remark this answers is not in the document]
Comment (ianjennings) [reply src/bootstrap.js]: 👍  Glad you tested

**Pull request merged**